packages/destination-actions/src/destinations/klaviyo/upsertProfile/index.ts

```diff
@@ -292,7 +297,12 @@ const action: ActionDefinition<Settings, Payload> = {
   },
 
   performBatch: async (request, { payload, hookOutputs }) => {
-    payload = payload.filter((profile) => profile.email || profile.external_id || profile.phone_number)
+    payload = payload.filter((profile) => {
+      if (profile.phone_number && !validatePhoneNumber(profile.phone_number)) {
+        return false
+      }
+      return profile.email || profile.phone_number || profile.external_id
```

Comment:
just a nit: you could probably do something like this and move the undefined check to validatePhoneNumber
```suggestion
      return profile.email || validatePhoneNumber(profile.phone_number) || profile.external_id
```

Reply:
I think this will keep the profiles which have invalid phone number but also has either email or external_id, we want to filter those out as well since Klaviyo will throw error for those too.